pkg/kncloudevents/retries.go

```diff
@@ -112,6 +119,15 @@ func RetryConfigFromDeliverySpec(spec v1.DeliverySpec) (RetryConfig, error) {
 		retryConfig.RequestTimeout, _ = timeout.Duration()
 	}
 
+	if spec.RetryAfterMax != nil {
```

Comment:
While this might be nice to do it is a bit problematic.  The goal was to do all validation of the experimental-feature in the Webhook in order to avoid requiring downstream implementations from having to load/watch the `config-features` ConfigMap into their context and then plumb that context through.  I suppose we could build a separate ConfigMap lookup inline here, but that feels heavyweight and non-standard with other ConfigMap tracking logic.  I'm inclined to not recheck the feature flags here but we can let others weigh in?

Reply:
I think you are right, and I didn't thought of other implementations.
On my side, this can be resolved.

Reply:
> While this might be nice to do it is a bit problematic. The goal was to do all validation of the experimental-feature in the Webhook in order to avoid requiring downstream implementations from having to load/watch the `config-features` ConfigMap into their context and then plumb that context through. I suppose we could build a separate ConfigMap lookup inline here, but that feels heavyweight and non-standard with other ConfigMap tracking logic. I'm inclined to not recheck the feature flags here but we can let others weigh in?

I think we actually **must** check if the feature is enabled. New experimental API fields will be stored as unknown fields if the feature is disabled and the webhook will ignore them. This means in a "enable-disable" scenario, this code will find a valid value for the `RetryAfterMax` and will enable it, even thought the feature is disabled. 

I believe experimental features are designed to be part of eventing core as:
1. API changes that're optionally implemented in alternative implementations.
2. A change in the reference implementation. This is to validate the feature, as well as provide a reference implementation for other alternative implementations. 

That means, that by design, if an alternative implementation will adopt an experimental feature, it needs to watch the features CM and provide its own implementation. Otherwise, it can wait until the graduation of the feature and becomes "on by default".